cli_tools/google-osconfig-agent/ospackage/ospackage.go

```diff
@@ -25,14 +26,78 @@ import (
 	"os"
 	"strings"
 
+	osconfig "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/cloud.google.com/go/osconfig/apiv1alpha1"
 	osconfigpb "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha1"
 	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/config"
 	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/logger"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/tasker"
+	"github.com/GoogleCloudPlatform/compute-image-tools/go/osinfo"
 	"github.com/GoogleCloudPlatform/compute-image-tools/go/packages"
+	"github.com/kylelemons/godebug/pretty"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc/status"
 )
 
-// SetConfig applies the configurations specified in the service.
-func SetConfig(res *osconfigpb.LookupConfigsResponse) error {
+var dump = &pretty.Config{IncludeUnexported: true}
+
+// Run looks up osconfigs and applies them.
+func Run(ctx context.Context, res string, enqueue bool) error {
+	client, err := osconfig.NewClient(ctx, option.WithEndpoint(config.SvcEndpoint()), option.WithCredentialsFile(config.OAuthPath()))
+	if err != nil {
+		return fmt.Errorf("osconfig.NewClient Error: %v", err)
+	}
+
+	resp, err := lookupConfigs(ctx, client, res)
+	if err != nil {
+		return fmt.Errorf("LookupConfigs error: %v", err)
+	}
+
+	if enqueue {
```

Comment:
shouldn't we always enqueue and know that the queue is empty and this will run right away if nothing else is happening?

Reply:
Right now we have no way to wait on tasks, once we do we can go this route as it would be cleaner